underscore.eachN.js

```diff
@@ -25,49 +25,83 @@
+        // The iterator is bound to the context object, if one is
+        // passed. Each invocation of iterator is called with three
+        // arguments: (slice, firstIndex, list), where firstIndex is
+        // the index of the beginning of the slice. If elements.length
+        // it not divisible by n, the last invocation to iterator will
+        // have fewer than n elements. Returns the original list for
+        // chaining.
+        eachN: function (obj, n, iterator, context) {
+
+            // Establish the object that gets returned to break out of a loop iteration.
+            var breaker = {};
+
+            if (typeof n !== 'number' || ! isFinite(n) || n % 1 !== 0) {
+                throw new TypeError('n should be an integer!');
+            }
+
+            if (obj === null) return obj;
+
+            for (var i = 0, length = obj.length; i < length; i += n) {
+                var slice = obj.slice(i, i + n);
+                if (iterator.call(context, slice, i, obj) === breaker) return;
+            }
+
+            return obj;
+        },
+
+        // Produces a new array of values by slicing list, n at a time,
+        // and mapping each slice through a transformation function
+        // (iterator).
+        mapN: function (obj, n, iterator, context) {
 
-        return obj;
-    },
+            var results = [];
+
+            if (obj === null) return results;
+
+            _.eachN(obj, n, function(slice, firstIndex, list) {
+                results.push(iterator.call(context, slice, firstIndex, list));
+            });
+
+            return results;
+        },
+
+        exports: function() {
+            var result = {};
+
+            for (var prop in this) {
+                if (!this.hasOwnProperty(prop)) continue;
+                result[prop] = this[prop];
+            }
+
+            return result;
+        }
+    };
+    // Exporting pattern copied from Underscore.string
+    // Exporting
 
-    // Produces a new array of values by slicing list, n at a time,
-    // and mapping each slice through a transformation function
-    // (iterator).
-    mapN: function (obj, n, iterator, context) {
+    // CommonJS module is defined
+    if (typeof exports !== 'undefined') {
+        if (typeof module !== 'undefined' && module.exports)
+            module.exports = _n;
 
-        var results = [];
+        exports._n = _n;
+    }
 
-        if (obj === null) return results;
+    // Register as a named module with AMD.
+    if (typeof define === 'function' && define.amd)
+        define('underscore.eachn', [], function(){ return _n; });
 
-        _.eachN(obj, n, function(slice, firstIndex, list) {
-            results.push(iterator.call(context, slice, firstIndex, list));
-        });
 
-        return results;
-    },
-});
+    // Integrate with Underscore.js if defined
+    // or create our own underscore object.
+    root._ = root._ || {};
+    root._.eachn = root._.eachn = _n;
```

Comment:
`_n` wraps up `eachN` and `mapN`. I would like to do `root._.n = root._.n = _n;'  so that we have `_.n.eachN` and `_.n.mapN`. This looks cleaner than `_.eachN.eachN` and `_.eachN.mapN`.